File: backend/PlugPlay/PlugPlay.Domain/Entities/Product.cs
namespace PlugPlay.Domain.Entities;

public class Product
{
    public int Id { get; set; }

    public int CategoryId { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public decimal Price { get; set; }

    public int StockQuantity { get; set; }

    public DateTime CreatedAt { get; set; }

    public Category Category { get; set; }

    public ICollection<ProductAttribute> ProductAttributes { get; set; } = new List<ProductAttribute>();

    public ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();

    public ICollection<Review> Reviews { get; set; } = new List<Review>();

    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/models/Product.ts
import {Review} from './Review';

export interface Product {
  id: number;
  name: string;
  price: number;
  description: string;
  stockQuantity: number;
  pictureUrls: string[];
  createdAt: string;
  category: {
    id: number;
    name: string;
    parentCategory?: {
      id: number;
      name: string;
    };
  } | null;
  reviews?: Review[];
}
